test/zeroD/test_zeroD.cpp

```diff
@@ -50,6 +51,185 @@ TEST(ZeroDim, test_individual_reactor_initialization)
+    sol->thermo()->setState_TPX(300.0, 101325, "CH4:1.0, O2:1.0, H2O:0, CO2:0");
+    // Set up reactor object
+    Reactor reactor;
+    reactor.insert(sol);
+    // Get reactions
+    auto kinetics = reactor.getKineticsMgr();
+    auto reaction0 = kinetics->getReaction(0);
+    auto reaction1 = kinetics->getReaction(1);
+    // Compare reaction strings to manual input
+    EXPECT_EQ(reaction0->equation(), "CH4 + 1.5 O2 => CO + 2 H2O");
+    EXPECT_EQ(reaction1->equation(), "CO + 0.5 O2 <=> CO2");
+}
+
+TEST(ZeroDim, test_get_thermo_from_reactor)
+{
+    // Setting up solution to insert in reactor
+    auto sol = newSolution("h2o2.yaml");
+    auto thermo1 = sol->thermo();
+    // Set up reactor object
+    Reactor reactor;
+    reactor.setThermoMgr(*thermo1);
+    // Get thermo from reactor
+    auto thermo2 = reactor.getThermoMgr();
+    // Compare two thermo objects
+    EXPECT_EQ(thermo1->nSpecies(), thermo2->nSpecies());
+    EXPECT_EQ(thermo1->enthalpy_mass(), thermo2->enthalpy_mass());
+    EXPECT_EQ(thermo1->intEnergy_mass(), thermo2->intEnergy_mass());
+    EXPECT_EQ(thermo1->pressure(), thermo2->pressure());
+}
+
+TEST(MoleReactorTestSet, test_preconditioned_ideal_const_vol)
+{
+    // Reactor 1
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TPX(1100.0, 10 * OneAtm, "H2:1.0, O2:0.5, AR:8.0");
+    IdealGasMoleReactor r1;
+    r1.insert(sol1);
+    // Reactor 2
+    auto sol2 = newSolution("h2o2.yaml");
+    sol2->thermo()->setState_TPX(1100.0, 10 * OneAtm, "H2:1.0, O2:0.5, AR:8.0");
+    IdealGasMoleReactor r2;
+    r2.insert(sol2);
+    // Network 1
+    ReactorNet net1;
+    net1.addReactor(r1);
+    AdaptivePreconditioner precon;
+    net1.setIntegratorType(&precon, GMRES);
+    net1.initialize();
+    // Network 2
+    ReactorNet net2;
+    net2.addReactor(r2);
+    net2.setIntegratorType(GMRES);
+    net2.initialize();
+    // Advancing
+    net1.advance(0.5);
+    net2.advance(0.5);
+    // Comparison
+    std::vector<double> state1(r1.neq(), 0.0);
+    std::vector<double> state2(r2.neq(), 0.0);
+    for (size_t i = 0; i < r1.neq(); i++)
+    {
+        EXPECT_NEAR(state1[i], state2[i], net1.atol());
+    }
+}
+
+TEST(MoleReactorTestSet, test_preconditioned_ideal_const_pressure)
+{
+    // Reactor 1
+    auto sol1 = newSolution("h2o2.yaml");
+    sol1->thermo()->setState_TPX(1100.0, 10 * OneAtm, "H2:1.0, O2:0.5, AR:8.0");
+    IdealGasConstPressureMoleReactor r1;
+    r1.insert(sol1);
+    // Reactor 2
+    auto sol2 = newSolution("h2o2.yaml");
+    sol2->thermo()->setState_TPX(1100.0, 10 * OneAtm, "H2:1.0, O2:0.5, AR:8.0");
+    IdealGasConstPressureMoleReactor r2;
+    r2.insert(sol2);
+    // Network 1
+    ReactorNet net1;
+    net1.addReactor(r1);
+    AdaptivePreconditioner precon;
+    net1.setIntegratorType(&precon, GMRES);
+    net1.initialize();
+    // Network 2
+    ReactorNet net2;
+    net2.addReactor(r2);
+    net2.setIntegratorType(GMRES);
```

Comment:
It is using SPGMR without preconditioned. It integrates just fine in this example. 

I did assume it was functional though because it was part of Cantera when I began this work.

https://github.com/Cantera/cantera/blob/032de5efa0412a870ed40cdf4125219c95956432/src/numerics/CVodesIntegrator.cpp#L393-L404